src/app/event/[id]/page.tsx

```diff
@@ -1,112 +1,70 @@
 "use client";
 
-import { GET_ATTESTATION_BY_ID_QUERY } from "@/lib/gqlEasAttestation/query";
-import { useQuery } from "urql";
-import { API_URL_MAPPING } from "@/lib/gqlEasAttestation";
-import { useAccount, useChainId } from "wagmi";
-import { useMemo } from "react";
-import { parseEventsData } from "../../events/parseEventsData";
+import { useAccount } from "wagmi";
 import { format } from "date-fns";
 
-import { Modal } from "../Modal";
-
 import {
   EventInfoContainer,
   StatusFlag,
-  Ticket,
-  TicketInfoContainer,
-  TicketInfoHeaderContainer,
+  TicketOwnerManagementSection,
 } from "../components/PageComponents";
 import { Calendar } from "lucide-react";
-import { Tabs, TabsContent, TabsList, TabsTrigger } from "@/components/ui/tabs";
-
-const ticketAddresses = [
-  { address: "0xkamscknsadkcnaksdncjsdncsjdn" },
-  { address: "0xkamscknsadkcnaksdncjsdncsjab" },
-  { address: "0xkamscknsadkcnaksdncjsdncsjac" },
-];
+import { useEventData } from "../hooks/useEventData";
+import { Address } from "viem";
+import { useTicketAddresses } from "../hooks/useTicketAddresses";
 
 export default function Page({ params }: { params: { id: string } }) {
   const eventId = params.id;
 
   const account = useAccount();
-  const chainId = useChainId();
 
-  const [result] = useQuery({
-    query: GET_ATTESTATION_BY_ID_QUERY,
-    variables: { id: eventId },
-    context: useMemo(() => ({ url: API_URL_MAPPING[chainId] }), [chainId]),
-  });
+  const {
+    eventData,
+    fetching: fetchingEvent,
+    error: errorEvent,
+  } = useEventData({ eventId });
 
-  const { data, fetching, error } = result;
-
-  const parsedData = useMemo(
-    () =>
-      data?.attestation && parseEventsData(data.attestation?.decodedDataJson),
-    [data?.attestation]
-  );
+  const { ticketAddresses } = useTicketAddresses({ eventId, eventData });
 
   if (!account) return <p>Please connect wallet to load the informations...</p>;
 
-  if (parsedData === undefined) return <p>loading...</p>;
+  if (eventData === undefined) return <p>loading...</p>;
```

Comment:
The loading should not be handled by  `fetchingEvent`? if so you should have something like that:

```suggestion
  if (fetchingEvent) return <p>loading...</p>;
```

Reply:
Well observed. I'm trying to solve here but it seems this system is not intelligent enough for this. So if I change this I will start getting ts warnings like "hey, eventData can be undefined"